src/root/hooks/useBooleanState.ts

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+
+type SetTrueFn = () => void;
+type SetFalseFn = () => void;
+type ToggleFn = () => void;
+type SetValueFn = React.Dispatch<React.SetStateAction<boolean>>;
+
+export default function useBooleanState(initialValue: boolean): [
+  boolean,
+  SetTrueFn,
+  SetFalseFn,
+  SetValueFn,
+  ToggleFn,
+] {
+  const [value, setValue] = React.useState(initialValue);
+
+  const setTrue = React.useCallback(() => {
+    setValue(true);
+  }, [setValue]);
+
+  const setFalse = React.useCallback(() => {
+    setValue(false);
+  }, [setValue]);
+
+  const toggleFn = React.useCallback(() => {
+    setValue((oldValue) => !oldValue);
+  }, [setValue]);
+
+  return [value, setTrue, setFalse, setValue, toggleFn];
```

Comment:
```suggestion
export default function useBooleanState(initialValue: boolean) {
  const [value, setValue] = React.useState(initialValue);

  const setTrue = React.useCallback(() => {
    setValue(true);
  }, [setValue]);

  const setFalse = React.useCallback(() => {
    setValue(false);
  }, [setValue]);

  const toggleFn = React.useCallback(() => {
    setValue((oldValue) => !oldValue);
  }, [setValue]);

  return [value, setTrue, setFalse, setValue, toggleFn] as const;
```

Reply:
Those return types was intentionally named to serve as documentation

Reply:
We don't need them as documentation. They are just redundant in this case IMO.

Reply:
I think i get it now. Let's return object instead of array.